cmd/oras/push.go

```diff
@@ -103,33 +104,45 @@ func runPush(opts pushOptions) error {
 	}
 
 	// specify resolver
-	var resolver remotes.Resolver
+	var (
+		resolver remotes.Resolver
+		ropts    *docker.ResolverOptions
+		err      error
+	)
 	if opts.dryRun {
 		resolver = iresolver.Dummy()
 		fmt.Println("Entered dry-run mode")
 	} else {
-		resolver = newResolver(opts.username, opts.password, opts.insecure, opts.plainHTTP, opts.configs...)
+		resolver, ropts = newResolver(opts.username, opts.password, opts.insecure, opts.plainHTTP, opts.configs...)
 	}
 
+	// load files
+	var (
+		annotations map[string]map[string]string
+		store       = content.NewFileStore("")
+	)
+
 	// bake artifact
 	var pushOpts []oras.PushOpt
 	if opts.artifactType != "" {
 		refResolver := resolver
 		if iresolver.IsDummy(resolver) {
-			refResolver = newResolver(opts.username, opts.password, opts.insecure, opts.plainHTTP, opts.configs...)
+			refResolver, _ = newResolver(opts.username, opts.password, opts.insecure, opts.plainHTTP, opts.configs...)
```

Comment:
Can we encapsulate this branching into the `newResolver` in resolve.go

Reply:
For this particular branch I can't because it would break --dry-run, but I can remove that call to newResolver